frmts/jp2kak/jp2kakdataset.cpp

```diff
@@ -2243,10 +2251,76 @@ JP2KAKCreateCopy( const char * pszFilename, GDALDataset *poSrcDS,
 
         CPLDebug("JP2KAK", "layer_bytes[] = %g\n",
                  static_cast<double>(layer_bytes[layer_count - 1]));
-    }
-    else
-    {
-        bReversible = true;
+    } else {
+        // Use RATE option.
+        char** dRateStr = CSLTokenizeStringComplex(CSLFetchNameValue(papszOptions,"RATE"),
```

Comment:
naming convention should be papszRate instead of dRateStr, and you'll need to CSLDestroy() it after use.
Other option
CPLStringList aosRate(CSLTokenizeStringComplex(CSLFetchNameValue(papszOptions,"RATE"),",", FALSE, FALSE ));
and it will get automatically freed.

Reply:
Fix in e134d20603ee19d0cd8af6db4d42345e6b34f866, 91764013b5f0600cd41dc84516e939039684b56d